### PR TITLE
Remove custom dimension from AB test configuration

### PR DESCRIPTION
**Pull request by hannako**
Universal Analytics has been replaced by Google Analytics 4. 

The custom dimension was a required field for UA, and is no longer needed by GA4 which processes events differently.

### CODE REVIEW

File: CHANGELOG.md
Comment (kevindew) [inline CHANGELOG.md:4]: Should we flag this as BREAKING? As dimension is part of the public API any code that initialises this class with dimension as an argument will now error.

File: README.md
Comment (kevindew) [inline README.md:32]: Do we still have value having the meta tag without the dimension? I've assumed (potentially incorrectly) that the primary purpose of the meta tag was for getting data into GA and not sure if that'd be problematic or superfluous without dimension.

**Review by kevindew — COMMENTED** [review]
Code looks good, I can't find any left references to dimension. I Did want to check that you're happy to do a breaking change.

As I'm not hugely familiar with the project I was a bit unclear what replaced the use of dimension now, if anything, and how things connect up to analytics after this? A link to the JS code could be useful to assert that.

**Comment by unoduetre** [comment]
As explained in the documentation in this repository, a special meta tag is used to pass AB testing information to Google Analytics. This meta tag has additional data attributes, one of them being data-analytics-dimension.

The old Universal Analytics code searched for the meta tag and got the dimension attribute from it. The code responsible for that has been removed, but it can be seen here: [app/assets/javascripts/govuk_publishing_components/analytics/custom-dimensions.js](https://github.com/alphagov/govuk_publishing_components/commit/c196da44f325896c617e0048075481eb8a6d42dc#diff-242c8a085773e8720119a395ea9590479f14970eb01892e8a495a1885b5fd17b) (line number 121).

The new GA4 code does not use this attribute anymore. The new code is here: 

* https://github.com/alphagov/govuk_publishing_components/blob/d0ff18b09e30923944614cadca5d67f5fdcea876/app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-page-views.js#L25
* https://github.com/alphagov/govuk_publishing_components/blob/d0ff18b09e30923944614cadca5d67f5fdcea876/app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-page-views.js#L152

File: README.md
Comment (unoduetre) [reply README.md:32]: Please, see my comment below.

File: CHANGELOG.md
Comment (unoduetre) [reply CHANGELOG.md:4]: Yes I think it makes sense to tag it as breaking and change the major version.

File: CHANGELOG.md
Comment (kevindew) [inline CHANGELOG.md:4]:
```suggestion
* Drop support for Ruby 3.0. The minimum required Ruby
  version is now 3.1
```

We don't have flag dropping Ruby versions as breaking changes: https://docs.publishing.service.gov.uk/manual/publishing-a-ruby-gem.html#ruby-version-compatibility

**Review by kevindew — APPROVED** [review]
Nice one, just a small tweak needed to the changelog but I don't think that'll need another review before merging.

File: CHANGELOG.md
Comment (unoduetre) [reply CHANGELOG.md]: Thank you for reviewing. I've removed the "breaking change" label from the line related to Ruby update and will merge it soon and then create the major version release PR.

**Pull request merged**
